File: app/src/main/java/com/ivanovsky/passnotes/util/InputMethodUtils.java
package com.ivanovsky.passnotes.util;

import android.app.Activity;
import android.content.Context;
import android.support.annotation.Nullable;
import android.view.View;
import android.view.inputmethod.InputMethodManager;

public class InputMethodUtils {

	public static void hideSoftInput(@Nullable Activity activity) {
		if (activity == null) return;

		View focusedView = activity.getWindow().getCurrentFocus();
		if (focusedView != null && focusedView.getWindowToken() != null) {
			InputMethodManager imm = (InputMethodManager) activity.getSystemService(Context.INPUT_METHOD_SERVICE);
			if (imm != null) {
				imm.hideSoftInputFromWindow(focusedView.getWindowToken(), 0);
			}
		}
	}

	public static void showSoftInput(Context context, View view) {
		if (context == null || view == null) return;

		view.requestFocus();
		view.postDelayed(() -> {
			InputMethodManager imm = (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE);
			if (imm != null) {
				imm.showSoftInput(view, InputMethodManager.SHOW_IMPLICIT);
			}
		}, 100);
	}

	private InputMethodUtils() {
	}
}

File: app/src/main/kotlin/com/ivanovsky/passnotes/util/InputMethodUtils.kt
package com.ivanovsky.passnotes.util

import android.app.Activity
import android.content.Context
import android.view.View
import android.view.inputmethod.InputMethodManager

object InputMethodUtils {
    @JvmStatic
    fun hideSoftInput(activity: Activity?) {
        val focusedView = activity?.window?.currentFocus ?: return
        val windowToken = focusedView.windowToken ?: return
        val inputMethodManager =
            activity.getSystemService(Context.INPUT_METHOD_SERVICE) as? InputMethodManager
        inputMethodManager?.hideSoftInputFromWindow(windowToken, 0)
    }

    @JvmStatic
    fun showSoftInput(context: Context?, view: View?) {
        if (context == null || view == null) return

        view.requestFocus()
        view.postDelayed(
            {
                val inputMethodManager =
                    context.getSystemService(Context.INPUT_METHOD_SERVICE) as? InputMethodManager
                inputMethodManager?.showSoftInput(view, InputMethodManager.SHOW_IMPLICIT)
            },
            100
        )
    }
}